src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx

```diff
@@ -45,6 +45,7 @@ export function ThermostatDeviceDetails({
   disableEditAccessCode,
   disableDeleteAccessCode,
   disableResourceIds = false,
+  disableScheduledClimates = false,
```

Comment:
These are internal components and should not have the default: the props should all be required. Actually the above default should be removed as well, and fixed in the lock device component too. Having these optional actually resulted in an earlier bug where the prop was not passed down.

```suggestion
  disableScheduledClimates
```

Reply:
Ahh, good point. I've removed this default specifier on all components except for `DeviceDetails` and `DeviceTable`. Are these the two top-level ones that should remain?

Reply:
Yes, generally, only defaults on top level components (or UI lib components) should be allowed 👍🏻

Reply:
Maybe we should do

```ts
interface ThermostatDeviceDetailsProps extends Required<Pick<CommonProps, '...'>> {
  device: ThermostatDevice
}

Reply:
I think this is a better paradigm, yeah. It makes it clearer why global props are being used in context.

Reply:
I think this should be done in a separate PR (right after this) for better encapsulation

Reply:
Ah, I remember why there was a default for `disableResourceIds`: the linter would complain about the nullable boolean. Your type change would fix that, but would be out of the original scope. I can keep it as-is with this default now, or replace the types on this component alone here.

I think I prefer the former, then creating a separate branch to make the type specificity change on all inner-component props. What do you think?

Either is fine—just thinking out loud!

Reply:
Ok to move it out of scope of this PR.